lib/Doctrine/ORM/UnitOfWork.php

```diff
@@ -3294,18 +3294,52 @@ private function performCallbackOnCachedPersister(callable $callback)
         }
     }
 
+    private function dispatchFlushEntityListener($event)
+    {
+        $subscribedSystems = [];
+        foreach ($this->entityChangeSets as $oid => $entityChangeSet) {
+            foreach ($this->identityMap as $className => $entities) {
+                $hash = (isset($this->entityIdentifiers[$oid], $this->entityIdentifiers[$oid]['id'])) ? $this->entityIdentifiers[$oid]['id'] : null;
+                if (!$hash) {
+                    continue;
+                }
+                $hash = (string) $hash;
+                if (!isset($this->identityMap[$className][$hash])) {
+                    continue;
+                }
+                $entity = $this->identityMap[$className][$hash];
+                if (spl_object_hash($entity) !== $oid) {
+                    continue;
+                }
+                $entityClass = get_class($entity);
+                $class = $this->em->getClassMetadata($entityClass);
+                if (!isset($subscribedSystems[$entityClass])) {
```

Comment:
CS not followed https://github.com/doctrine/doctrine2/blob/master/CONTRIBUTING.md#coding-standard